### PR TITLE
ARROW-15667: [R] Test development build with ARROW_BUILD_STATIC=OFF

### PR DESCRIPTION
**Pull request by wjones127**
I found that on Windows, the build would fail if `ARROW_BUILD_STATIC=OFF`, even though it was supposed to be using shared libraries.

To make things simpler, I changed `configure.win` to just use the pkg-config files. This works for dynamic libraries. However, if you try to do a static-only build, it fails. I started to look into why, but am feeling exhausted debugging linking errors. 😩  At the very least, I think this change means if someone fixes the pkg-config file to support static linking, then the R static-only build should (hopefully) be fixed automatically.

### CODE REVIEW

**Comment by wjones127** [comment]
@github-actions crossbow submit test-r-devdocs

**Comment by wjones127** [comment]
@github-actions crossbow submit test-r-devdocs

**Comment by pitrou** [comment]
What's the status on this? It seems you need to link to specific system libraries for mimalloc?

**Comment by wjones127** [comment]
> What's the status on this? It seems you need to link to specific system libraries for mimalloc?

Yes, this should nearly be finished. I should get this ready for review tomorrow.

**Comment by wjones127** [comment]
@github-actions crossbow submit test-r-devdocs

**Comment by wjones127** [comment]
@github-actions crossbow submit test-r-devdocs

**Comment by wjones127** [comment]
@github-actions crossbow submit test-r-devdocs

**Comment by wjones127** [comment]
@github-actions crossbow submit test-r-devdocs

**Comment by wjones127** [comment]
@github-actions crossbow submit test-r-devdocs

**Comment by wjones127** [comment]
Crossbow build now passing. I believe the two CI failures are unrelated (segfault in hash join C++ tests and io error in S3 C++ tests).

**Comment by wjones127** [comment]
Thanks for the review @kou. Those suggests all make sense.

**Comment by wjones127** [comment]
@github-actions crossbow submit test-r-devdocs

File: r/vignettes/developers/setup.Rmd [hunk 1]
```diff
@@ -110,13 +110,11 @@ The package can be built on Windows using [RTools 4](https://cran.r-project.org/
 
 Open the corresponding RTools Bash, for example "Rtools MinGW 64-bit" for mingw64.
 
-Install CMake and Ninja with:
+Install CMake, CCache, and Ninja with:
```
Comment (kou) [inline r/vignettes/developers/setup.Rmd:113]:
```suggestion
Install CMake, ccache, and Ninja with:
```

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**